app/controllers/checkout_controller.rb

```diff
@@ -24,13 +24,21 @@ def edit
   def update
     return update_failed unless @order.update_attributes(object_params)
 
+    shipping_method_id = object_params[:shipping_method_id]
+    object_params.delete(:shipping_method_id)
```

Comment:
Why do we delete that entry from the parameters? It has already been passed on two lines above.

Reply:
ah, it's a mistake. I need to add unit tests to this and make sure it works.
We will delete shipping_method_id before calling update_attributes because order.shipping_method_id will not be in the model (we will drop the column afterwards).

Even if ActiveRecord can handle inexistent attributes on update_attributes (I am not sure about this) it's better to keep this explicit: the difference between UI code (object_params) where order.shipping_method_id is used and DB/AR code (@order.update_attributes) where order.shipping_method_id is not present.